conans/client/graph/graph_binaries.py

```diff
@@ -40,6 +41,8 @@ def __init__(self, conan_app, global_conf):
         python_mode = global_conf.get("core.package_id:default_python_mode", default="minor_mode")
         build_mode = global_conf.get("core.package_id:default_build_mode", default=None)
         self._modes = unknown_mode, non_embed, embed_mode, python_mode, build_mode
+        self._hook_manager = HookManager(HomePaths(self._home_folder).hooks_path)
```

Comment:
This can make the batch processing at scale in repos like ConanCenter very slow, as this will be instantiated tons of times. Need to be factored out from here (this might be complicated and require some re-architecture)

cc @AbrilRBS

Reply:
I just need to access the Conan App to capture the hook paths, then, instantiate the hook manager. It would be great doing it in `run_validate_package_id` directly, but I have no access to Conan App. Is there a way to capture the Conan App globally? I also don't like this current approach of forwarding the hook manager til run_validate_package_id

Reply:
Or, if you make it a class instance, you can save future HookLoads for other places in the code! But then you'll have issues with the tests keping states between each call so, pick your poison I guess